scripts/run-integration-tests.sh

```diff
@@ -210,50 +212,35 @@ grep -r -q $TEST_IMAGE_VERSION $TEST_CONFIG_PATH
-echo "Running ginkgo tests with focus: $focus"
-(cd "$INTEGRATION_TEST_DIR/cni" && CGO_ENABLED=0 ginkgo --focus="$focus" -v --timeout 20m --no-color --fail-on-pending -- --cluster-kubeconfig="$KUBECONFIG" --cluster-name="$CLUSTER_NAME" --aws-region="$AWS_DEFAULT_REGION" --aws-vpc-id="$VPC_ID" --ng-name-label-key="kubernetes.io/os" --ng-name-label-val="linux")
-(cd "$INTEGRATION_TEST_DIR/ipamd" && CGO_ENABLED=0 ginkgo --focus="$focus" -v --timeout 10m --no-color --fail-on-pending -- --cluster-kubeconfig="$KUBECONFIG" --cluster-name="$CLUSTER_NAME" --aws-region="$AWS_DEFAULT_REGION" --aws-vpc-id="$VPC_ID" --ng-name-label-key="kubernetes.io/os" --ng-name-label-val="linux")
-TEST_PASS=$?
-DEFAULT_INTEGRATION_DURATION=$((SECONDS - START))
-echo "TIMELINE: Default CNI integration tests took $DEFAULT_INTEGRATION_DURATION seconds."
+    echo "*******************************************************************************"
+    echo "Running integration tests on default CNI version, $ADDONS_CNI_IMAGE"
+    echo ""
+    START=$SECONDS
+
+    focus="CANARY"
+    echo "Running ginkgo tests with focus: $focus"
+    (cd "$INTEGRATION_TEST_DIR/cni" && CGO_ENABLED=0 ginkgo --focus="$focus" -v --timeout 60m --no-color --fail-on-pending -- --cluster-kubeconfig="$KUBECONFIG" --cluster-name="$CLUSTER_NAME" --aws-region="$AWS_DEFAULT_REGION" --aws-vpc-id="$VPC_ID" --ng-name-label-key="kubernetes.io/os" --ng-name-label-val="linux")
+    (cd "$INTEGRATION_TEST_DIR/ipamd" && CGO_ENABLED=0 ginkgo --focus="$focus" -v --timeout 60m --no-color --fail-on-pending -- --cluster-kubeconfig="$KUBECONFIG" --cluster-name="$CLUSTER_NAME" --aws-region="$AWS_DEFAULT_REGION" --aws-vpc-id="$VPC_ID" --ng-name-label-key="kubernetes.io/os" --ng-name-label-val="linux")
+    TEST_PASS=$?
+    DEFAULT_INTEGRATION_DURATION=$((SECONDS - START))
+    echo "TIMELINE: Default CNI integration tests took $DEFAULT_INTEGRATION_DURATION seconds."
+fi
 
 echo "*******************************************************************************"
 echo "Updating CNI to image $IMAGE_NAME:$TEST_IMAGE_VERSION"
 echo "Using init container $INIT_IMAGE_NAME:$TEST_IMAGE_VERSION"
 START=$SECONDS
 $KUBECTL_PATH apply -f "$TEST_CONFIG_PATH"
-NODE_COUNT=$($KUBECTL_PATH get nodes --no-headers=true | wc -l)
-echo "Number of nodes in the test cluster is $NODE_COUNT"
-UPDATED_PODS_COUNT=0
-MAX_RETRIES=20
-RETRY_ATTEMPT=0
-sleep 5
-while [[ $UPDATED_PODS_COUNT -lt $NODE_COUNT && $RETRY_ATTEMPT -lt $MAX_RETRIES ]]; do
-    UPDATED_PODS_COUNT=$($KUBECTL_PATH get pods -A --field-selector=status.phase=Running -l k8s-app=aws-node --no-headers=true | wc -l)
-    let RETRY_ATTEMPT=RETRY_ATTEMPT+1
-    sleep 5
-    echo "Waiting for cni daemonset update. $UPDATED_PODS_COUNT are in ready state in retry attempt $RETRY_ATTEMPT"
-done
-echo "Updated!"
-
-CNI_IMAGE_UPDATE_DURATION=$((SECONDS - START))
-echo "TIMELINE: Updating CNI image took $CNI_IMAGE_UPDATE_DURATION seconds."
+check_ds_rollout "aws-node" "kube-system" "4m"
```

Comment:
Wondering if 4min timeout is sufficient as there could be upto 6 `aws-node` pods in the test cluster. 
Let's trigger an integration test before merging the PR.

Reply:
Good call, updated to 10m, which I think should be sufficient for now